lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,29 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n x m) where n is the length of the matrix and m is the length of the longest array in the matrix.
+# In the first loop, the outer loop will run n times and the inner loop will run m times. The second loop will run n times.
+# Space complexity: O(n) where n is the length of the matrix because the number of key/value pairs will be equal to the number
+# of indices in the matrix.
+
 def matrix_check_sum(matrix)
-  raise NotImplementedError
+  sums = {}
+
+  matrix.each do |arr|
+    arr.each_with_index do |num, j|
+      if !sums.has_key?(j)
```

Comment:
This works, but you are only recording the sums of each column here and then the following loop finds the sum of each row.  Could you use the indices to calculate both at the same time?

If you did so, you could reduce the number of loops and reduce the space complexity.